lib/dry/schema/extensions/info/schema_compiler.rb

```diff
@@ -83,7 +83,14 @@ def visit_each(node, opts = EMPTY_HASH)
         # @api private
         def visit_key(node, opts = EMPTY_HASH)
           name, rest = node
-          visit(rest, opts.merge(key: name, required: true))
+          visit(rest, opts.merge(key: name, required: true, nullable: false))
+        end
+
+        # @api private
+        def visit_not(_node, opts = EMPTY_HASH)
+          key = opts[:key]
+
+          keys[key][:nullable] = true
         end
```

Comment:
Sorry, I don't have time for a deeper look but this analysis seems very shallow. You cannot deduce a key is nullable from merely using `not`, this is too general. Of course, the tests pass but it's only because there are too few of them. I think you ought to analyze in `visit_predicate`, this is the place where you can detect if it was the maybe macro.

Reply:
Sure, will do that 💯